### PR TITLE
Build memcached extension debian package

### PR DESCRIPTION
**Pull request by tmatsuo**
@bshaffer 
@dazuma

I'd appreciate it if you could review it.

For now I manually conformed that the debian packages are correctly built for PHP 5.6.x and PHP 7.0.x.

### CODE REVIEW

**Comment by tmatsuo** [comment]
#177 is using the debian packages built by this PR and the tests are passing :)

File: deb-package-builder/build.sh [hunk 1]
```diff
@@ -65,11 +68,25 @@ for FULL_VERSION in $(echo ${PHP_VERSIONS} | tr "," "\n"); do
             gcp-php-${PHP_VERSION}-${PHP_VERSION}
         popd
     fi
-    sed -i -e "s/PHP_VERSION/${PHP_VERSION}/" debian/control debian/rules
-    sed -i -e "s/SHORT_VERSION/${SHORT_VERSION}/" debian/control debian/rules \
-        debian/patches/series
+    envsubst '${PHP_VERSION},${SHORT_VERSION}' < debian/rules > debian/rules.tmp
```
Comment (dazuma) [inline deb-package-builder/build.sh]: Consider naming the original (template) file something like `debian/rules.in`. Same with the control and patches files below.

File: deb-package-builder/build.sh [hunk 2]
```diff
@@ -65,11 +68,25 @@ for FULL_VERSION in $(echo ${PHP_VERSIONS} | tr "," "\n"); do
...
         "Build ${FULL_VERSION} of gcp-php-${PHP_VERSION}"
-    dpkg-buildpackage -us -uc
+    dpkg-buildpackage -us -uc -j"$(nproc)"
+    popd
+    # build extensions
+    dpkg -i gcp-php-${PHP_VERSION}_${FULL_VERSION}_amd64.deb
+    # Make it a default
+    rm -rf ${PHP_DIR}
+    ln -sf /opt/php${SHORT_VERSION} ${PHP_DIR}
+    which php
```
Comment (dazuma) [inline deb-package-builder/build.sh]: What is this for? Is it just to fail if the binary was not properly installed?

File: deb-package-builder/extensions/memcached/build.sh [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    rm -rf memcached/.git
+    mv memcached ${PACKAGE_DIR}
+    tar -cvzf ${PNAME}-${PACKAGE_VERSION}.orig.tar.gz ${PACKAGE_DIR}
+elif [ ${SHORT_VERSION} == '71' ]; then
+    echo "Not yet implemented"
+    exit 0
+fi
+
+cp -R ${DEB_BUILDER_DIR}/extensions/memcached/debian ${PACKAGE_DIR}
+envsubst '${SHORT_VERSION}' < ${PACKAGE_DIR}/debian/rules \
```
Comment (dazuma) [inline deb-package-builder/extensions/memcached/build.sh]: Similar: consider naming the template `$PACKAGE_DIR/debian/rules.in`

File: deb-package-builder/build.sh (hunk 2 above)
Comment (tmatsuo) [inline deb-package-builder/build.sh]: Good catch, removed :)

File: deb-package-builder/extensions/memcached/build.sh (hunk 3 above)
Comment (tmatsuo) [inline deb-package-builder/extensions/memcached/build.sh]: Done

**Comment by tmatsuo** [comment]
@dazuma Thanks for the review! PTAL

**Comment by tmatsuo** [comment]
Closing in favor of #178

**Pull request closed without merging**